File: medical-beauty-app/server/src/main/java/com/starry/mb/plan/domain/Plan.java
package com.starry.mb.plan.domain;

import com.baomidou.mybatisplus.annotation.TableName;
import com.starry.mb.common.domain.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@EqualsAndHashCode(callSuper = true)
@TableName("plan")
public class Plan extends BaseEntity {
    private Long customerId;
    private String title;
    private String subtitle;
    private String coverUrl;
    private Long consultantId;
    private LocalDate validFrom;
    private LocalDate validTo;
    private BigDecimal totalPrice;
    private BigDecimal discountPrice;
    private String analysisText;
    private String analysisPhotoKeys;
    private Integer status;
    private Integer version;
    private LocalDateTime pushedAt;
}

File: medical-beauty-app/admin/src/api/plan.ts
import client from './client';

export interface PlanSection {
  id?: number;
  sort?: number;
  type: string;
  title: string;
  content?: string;
}
export interface PlanItem {
  id?: number;
  sectionId?: number;
  projectId: number;
  projectName?: string;
  plannedCount?: number;
  doneCount?: number;
  unitPrice?: string | number;
  activityPrice?: string | number;
  totalPrice?: string | number;
  status?: number;
  notes?: string;
}
export interface Plan {
  id?: number;
  customerId: number;
  title: string;
  subtitle?: string;
  coverUrl?: string;
  validFrom?: string;
  validTo?: string;
  totalPrice?: string | number;
  discountPrice?: string | number;
  analysisText?: string;
  status?: number;
  version?: number;
  pushedAt?: string;
}
export interface PlanDetail {
  plan: Plan;
  sections: PlanSection[];
  items: PlanItem[];
}
export interface PlanUpsert extends Plan {
  sections: PlanSection[];
  items: PlanItem[];
}

export const listPlansByCustomer = (customerId: number) =>
  client.get<Plan[], Plan[]>('/api/admin/plans', { params: { customerId } });

export const getPlan = (id: number) =>
  client.get<PlanDetail, PlanDetail>(`/api/admin/plans/${id}`);

export const createPlan = (data: PlanUpsert) =>
  client.post<Plan, Plan>('/api/admin/plans', data);

export const updatePlan = (id: number, data: PlanUpsert) =>
  client.put<Plan, Plan>(`/api/admin/plans/${id}`, data);

export const pushPlan = (id: number) =>
  client.post<Plan, Plan>(`/api/admin/plans/${id}/push`);

export const deletePlan = (id: number) =>
  client.delete<void, void>(`/api/admin/plans/${id}`);
